### PR TITLE
Add intersection and union for selections

### PR DESCRIPTION
**Pull request by mgeplf**
* this is in preparation for handling node_sets

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -11,5 +11,4 @@ target_link_libraries(unittests
     sonata_shared
     Catch2::Catch2
 )
-
```
Comment (matz-e) [inline tests/CMakeLists.txt]: really?

File: include/bbp/sonata/population.h [hunk 2]
```diff
@@ -12,9 +12,10 @@
 #include "common.h"
 
 #include <cstdint>
-#include <memory>
+#include <memory>  // std::unique_ptr, std::unique_ptr
```
[the remark this answers is not in the document]
Comment (mgeplf) [reply include/bbp/sonata/population.h]: sigh, fixed

File: include/bbp/sonata/population.h [hunk 3]
```diff
@@ -38,6 +39,9 @@ class SONATA_API Selection
     static Selection fromValues(Iterator first, Iterator last);
     static Selection fromValues(const Values& values);
 
+    Selection intersection_(const Selection&) const;
+    Selection union_(const Selection&) const;
+
```
[the remark this answers is not in the document]
Comment (mgeplf) [reply include/bbp/sonata/population.h]: good point, I did one better, removed them and put them in `detail::`, makes way more sense.

File: tests/CMakeLists.txt (hunk 1 above)
Comment (mgeplf) [reply tests/CMakeLists.txt]: yup

**Pull request merged**